aws/credentials/credentials.go

```diff
@@ -1,61 +1,75 @@
 package credentials
 
 import (
-	"net/http"
+	"context"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
-	"github.com/aws/aws-sdk-go/aws/ec2metadata"
-	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/transport/http"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 )
 
 type AWSCredentials interface {
-	Get() (*credentials.Credentials, error)
+	Get() (*aws.Credentials, error)
+	GetCredentialsProvider() (aws.CredentialsProvider, error)
 }
 
 type awsCred struct {
-	creds *credentials.Credentials
+	creds         *aws.Credentials
+	credsprovider aws.CredentialsProvider
 }
 
 func NewAWSCredentials(id, secret, token string, runningOnEc2 bool) (AWSCredentials, error) {
-	var creds *credentials.Credentials
+	var creds aws.Credentials
+	var credsprovider aws.CredentialsProvider
+	var ctx context.Context
 	if id != "" && secret != "" {
-		creds = credentials.NewStaticCredentials(id, secret, token)
-		if _, err := creds.Get(); err != nil {
+		cfg, err := config.LoadDefaultConfig(ctx, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(id, secret, token)))
+		if err != nil {
 			return nil, err
 		}
-	} else {
-		providers := []credentials.Provider{
-			&credentials.EnvProvider{},
+
+		creds, err = cfg.Credentials.Retrieve(context.Background())
+		if err != nil {
+			return nil, err
 		}
-		if runningOnEc2 {
-			client := http.Client{Timeout: time.Second * 10}
-			sess := session.Must(session.NewSession())
-			ec2RoleProvider := &ec2rolecreds.EC2RoleProvider{
-				Client: ec2metadata.New(sess, &aws.Config{
-					HTTPClient: &client,
-				}),
-			}
-			providers = append(providers, ec2RoleProvider)
+
+	} else if runningOnEc2 {
```

Comment:
Curious, does AWS not allow chaining of providers anymore?  Previously we were able to add EnvProvider and RoleProvider together. With this change we won't since each provider is in its own if condition.